File: src/main/java/br/com/g1bet/model/Aposta.java
package br.com.g1bet.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
public class Aposta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_usuario")
    private Usuario usuario;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_partida")
    private Partida partida;

    @NotNull
    @Column(name = "tipo_de_aposta")
    @Enumerated(EnumType.STRING)
    private TipoApostaEnum tipo;

    @NotNull
    private Double valorApostado;

    @NotNull
    private Double odd = (Math.random() * 5);

    public Aposta() {

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public Partida getPartida() {
        return partida;
    }

    public void setPartida(Partida partida) {
        this.partida = partida;
    }

    public TipoApostaEnum getTipo() {
        return tipo;
    }

    public void setTipo(TipoApostaEnum tipo) {
        this.tipo = tipo;
    }

    public Double getValorApostado() {
        return valorApostado;
    }

    public void setValorApostado(Double valorApostado) {
        this.valorApostado = valorApostado;
    }

    public Double getOdd() {
        return odd;
    }

    public void setOdd(Double odd) {
        this.odd = odd;
    }

}

File: src/main/java/br/com/g1bet/model/Aposta.kt
package br.com.g1bet.model

import javax.persistence.*
import javax.validation.constraints.NotNull

@Entity
class Aposta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne(cascade = arrayOf(CascadeType.ALL))
    @JoinColumn(name = "id_usuario")
    var usuario: Usuario? = null

    @ManyToOne(cascade = arrayOf(CascadeType.ALL))
    @JoinColumn(name = "id_partida")
    var partida: Partida? = null

    @javax.validation.constraints.NotNull
    @Column(name = "tipo_de_aposta")
    @Enumerated(EnumType.STRING)
    var tipo: TipoApostaEnum? = null

    @javax.validation.constraints.NotNull
    var valorApostado: Double? = null

    @javax.validation.constraints.NotNull
    var odd = Math.random() * 5
        get() = arredondarParaDuasCasas(field)

    companion object {
        fun arredondarParaDuasCasas(d: Double): Double {
            return (if (d < 0) d * 100 - 0.5 else d * 100 + 0.5).toLong() / 100.0
        }
    }
}